pkg/sql/colenc/colenc_test: TestEncoderEqualityRand failed

pkg/sql/colenc/colenc_test.TestEncoderEqualityRand [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/19613618?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/19613618?buildTab=artifacts#/) on release-23.2 @ [822b048895af77a237f1da6fb361b658871d4b52](https://github.com/cockroachdb/cockroach/commits/822b048895af77a237f1da6fb361b658871d4b52):


```
=== RUN   TestEncoderEqualityRand
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/4e051761860c8ed717bdc3d4d37b7430/logTestEncoderEqualityRand3670535788
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:154: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    sql_runner.go:83: error executing query="CREATE TABLE t97 (col97_0 BYTES NOT NULL, \"col9%p7_1\" \"char\" NULL, col😑97_2 BOOL NOT NULL, col97_3 STRING NULL AS (lower(\"col9%p7_1\")) STORED, col9😦7_4 STRING NOT NULL AS (CASE WHEN col97_0 IS NULL THEN e'W\\x0bR':::STRING ELSE 'Y':::STRING END) STORED, INDEX (col9😦7_4, (CASE WHEN col97_0 IS NULL THEN e'\\x04Evk':::STRING ELSE '':::STRING END)), INDEX (col9😦7_4, \"col9%p7_1\" ASC, col😑97_2, col97_0 ASC, lower(\"col9%p7_1\")) STORING (col97_3), INDEX (col97_3 ASC, (CASE WHEN col97_0 IS NULL THEN e'v-i\\t1oE':::STRING ELSE 'CWh[cp^w5':::STRING END) ASC) STORING (col9😦7_4) NOT VISIBLE, INVERTED INDEX (col97_0 ASC, col😑97_2, \"col9%p7_1\" gin_trgm_ops ASC), INDEX (col97_0 ASC) PARTITION BY LIST (col97_0) (PARTITION t97_p̟art0 VALUES IN (('\\x7445c3b4df79ac':::BYTES,), ('\\x58':::BYTES,), ('\\xaebda4ee37e49d':::BYTES,), ('\\x':::BYTES,), ('\\x6ebab7e8b169':::BYTES,)), PARTITION t97_part1 VALUES IN (('\\xf57bd96119594d1a':::BYTES,), ('\\xd3ed9d46':::BYTES,), ('\\x1dbdc0fdaf7597e60e':::BYTES,), ('\\xd1ad385146e081':::BYTES,)), PARTITION \"t97\"\"_part2\" VALUES IN (('\\xd7687ce6a4':::BYTES,), ('\\x77c7a60b781de185':::BYTES,), ('\\xac21ba900108a5':::BYTES,), ('\\x9a1062':::BYTES,)), PARTITION t97_part3 VALUES IN (('\\x8c252538':::BYTES,), ('\\x94dcc6ada947':::BYTES,), ('\\x22d9ad4177797d':::BYTES,), ('\\xe96ee575f14ad98bc4':::BYTES,)), PARTITION t97_part4 VALUES IN (('\\xcd85':::BYTES,), ('\\x6311':::BYTES,), ('\\xeea9':::BYTES,), ('\\xcaf38c84cbbe':::BYTES,)), PARTITION \"t97_%vpart5\" VALUES IN (('\\x5bfdf00d789fe383':::BYTES,), ('\\x00':::BYTES,), ('\\xcefb493c2c70b957a4':::BYTES,), ('\\xd877a56816fe':::BYTES,), ('\\x0448b76af95821':::BYTES,)), PARTITION t97_part6 VALUES IN (('\\x0657c3':::BYTES,), ('\\x223765a9':::BYTES,), ('\\x3f64':::BYTES,), ('\\x1761917106a2':::BYTES,), ('\\xa5eea63c881e95433f':::BYTES,)), PARTITION 🙁t97_part7 VALUES IN (('\\x22':::BYTES,), ('\\x470b69a9722c':::BYTES,), ('\\x1001f90dbcbb':::BYTES,), ('\\xef3c927a20271cdfd8':::BYTES,), ('\\x3774a047f38f25d4':::BYTES,)), PARTITION t97_part8 VALUES IN (('\\x203faed8ce24d5b8':::BYTES,), ('\\xce24d0e893dfecb74a':::BYTES,), ('\\xdcbc01b0df0d0f':::BYTES,), ('\\xe29883':::BYTES,))), UNIQUE (\"col9%p7_1\" DESC) STORING (col97_0, col😑97_2), UNIQUE (\"col9%p7_1\" ASC, col😑97_2 ASC) STORING (col97_0, col97_3), INDEX (col97_0) PARTITION BY LIST (col97_0) (PARTITION t97_part0 VALUES IN (('\\xead0':::BYTES,), ('\\x14ddcec6983e74bf52':::BYTES,), ('\\x5ae3b0':::BYTES,), ('\\x6b5c3e59':::BYTES,), ('\\x':::BYTES,)), PARTITION \"t97\\f_part1\" VALUES IN (('\\xa3c196d0528fce38':::BYTES,), ('\\x847d7b02202b3a':::BYTES,), ('\\x22':::BYTES,), ('\\x2d':::BYTES,), ('\\x79da4fc1':::BYTES,)), PARTITION \"t97_\\vpart2\" VALUES IN (('\\x78429b172e':::BYTES,), ('\\x46a874ac':::BYTES,), ('\\x60f2':::BYTES,), ('\\xee3f4b236559c2':::BYTES,)), PARTITION t97_part3 VALUES IN (('\\x0fdc3103cb8741':::BYTES,), ('\\x9b355857a2a8':::BYTES,), ('\\xbe4fe8bd':::BYTES,), ('\\x00':::BYTES,), ('\\x57c8f8b55c':::BYTES,)), PARTITION t97_part4 VALUES IN (('\\xd33d9abcfb0d9674':::BYTES,), ('\\xa012':::BYTES,), ('\\xc54ef8d9d3903a':::BYTES,), ('\\xb9ea41fa8c':::BYTES,), ('\\x0687c702':::BYTES,)), PARTITION \"t\"\"97_part 5\" VALUES IN (('\\x4c91a398fb':::BYTES,), ('\\xb5a34d480157975499':::BYTES,), ('\\x7f1b19':::BYTES,), ('\\x3857c83edd9b':::BYTES,), ('\\x568751fbfd':::BYTES,)), PARTITION \"DEFAULT\" VALUES IN ((DEFAULT,))), UNIQUE (col97_3 ASC, \"col9%p7_1\" DESC, col😑97_2, col97_0 DESC, col9😦7_4 ASC), FAMILY (col97_3, \"col9%p7_1\"), FAMILY (col😑97_2), FAMILY (col97_0, col9😦7_4))" args=[]: pq: exceeded limit for number of table spans
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/4e051761860c8ed717bdc3d4d37b7430/logTestEncoderEqualityRand3670535788
--- FAIL: TestEncoderEqualityRand (28.79s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestEncoderEqualityRand.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-50364

--- a/pkg/sql/colenc/encode_test.go
+++ b/pkg/sql/colenc/encode_test.go
@@ -259,6 +259,10 @@ func TestEncoderEqualityRand(t *testing.T) {
 	ctx := context.Background()
 	s, db, kvdb := serverutils.StartServer(t, base.TestServerArgs{})
 	defer s.Stopper().Stop(ctx)
+	// Increase the span config limit in case we're running with multiple
+	// tenants since the loop below might create more spans than the default
+	// limit of 5k.
+	s.SQLConn(t).QueryRow("SET CLUSTER SETTING spanconfig.tenant_limit = 50000")
 	codec, sv := s.ApplicationLayer().Codec(), &s.ApplicationLayer().ClusterSettings().SV
 	rng, _ := randutil.NewTestRand()
 	for i := 0; i < 100; i++ {

